ccm_web/client/src/pages/FormatThirdPartyGradebook.tsx

```diff
@@ -0,0 +1,446 @@
+            <Grid container>
+              <Grid item xs={12}>
+                <CircularProgress color='inherit' />
+              </Grid>
+              <Grid item xs={12}>
+                Loading
+                {isGetSectionsLoading && ' section '}
+                {isGetStudentsLoading && ' student '}
+                data from Canvas
+              </Grid>
+            </Grid>
+          </Backdrop>
+        </div>
+        <Grid container className={classes.buttonGroup} justify='flex-end'>
+          <Button
+            color='primary'
+            variant='contained'
+            disabled={selectedSections === undefined}
+            aria-label='Select section'
+            onClick={handleSelectClick}
+          >
+            Select{selectedSections !== undefined ? ` (${selectedSections.length})` : ''}
+          </Button>
+        </Grid>
+      </div>
+    )
+  }
+
+  const renderUpload = (): JSX.Element => {
+    if (schemaInvalidations !== undefined) {
+      return renderSchemaInvalidations(schemaInvalidations)
+    }
+    if (gradebookInvalidations !== undefined) {
+      return renderGradebookInvalidations(gradebookInvalidations)
+    }
+
+    const description = (
+      'This tool creates a new version of the uploaded file so it only includes students in the selected sections ' +
+      'and is formatted appropriately for uploading to Canvas.'
+    )
+    const gradeUploadDocsLink = 'https://community.canvaslms.com/t5/Instructor-Guide/tkb-p/Instructor#Grades'
+    const requirements = (
+      <ol>
+        <li>
+          <Typography>
+            The file includes a &quot;{REQUIRED_LOGIN_ID_HEADER}&quot; column (the Canvas equivalent of Uniqname)
+            and one other column of scores, with the column&apos;s header being the name of a new assignment.
+            Other columns required by Canvas
+            (see the <Link href={gradeUploadDocsLink} target='_blank' rel='noopener'>help docs</Link>)
+            are allowed but not required.
+          </Typography>
+        </li>
+        <li>
+          <Typography>
+            The first non-header row is a &quot;Points Possible&quot; row, meaning that the value in the new
+            assignment column will be the maximum points possible for that assignment
+            and that &quot;Points Possible&quot; is present in another cell in the row
+            (such as the row&apos;s cell for &quot;{REQUIRED_LOGIN_ID_HEADER}&quot;).
+          </Typography>
+        </li>
+        <li>
+          <Typography>
+            Your file must include records for at least one student in the selected section
+            (students are identified using the &quot;{REQUIRED_LOGIN_ID_HEADER}&quot; value).
+          </Typography>
+        </li>
+      </ol>
+    )
+    const fileData = (
+      'SIS Login ID,Example Assignment\n' +
+      'Points Possible,100\n' +
+      'studentone,80\n' +
+      'studenttwo,90\n'
+    )
```

Comment:
Optional: I feel these sample files should be coming from utility file so when changes are requested we know where to go.

Reply:
I was following the existing pattern. I think this could be a separate issue.

Reply:
Yeah... Not now.